include/grpc/byte_buffer.h

```diff
@@ -34,10 +34,17 @@
 #ifndef GRPC_BYTE_BUFFER_H
 #define GRPC_BYTE_BUFFER_H
 
-#include <grpc/grpc.h>
 #include <grpc/support/slice_buffer.h>
 
-typedef enum { GRPC_BB_SLICE_BUFFER } grpc_byte_buffer_type;
+typedef enum {
+  GRPC_BB_SLICE_BUFFER,
+
+  /* Keep the GRPC_BB_COMPRESSED_* entries in the same order as the
+   * grpc_compression_algorithm enum entries. */
+  GRPC_BB_COMPRESSED_NONE, /* for overriding otherwise compressed channels */
```

Comment:
The meaning behind the comment was that a channel compressed by default would, when instructed, not compress certain messages, resulting in said messages having this BB_COMPRESSED_NONE type. This would signal that the message indeed came from a compressing source but that in this instance, the data isn't compressed. I wonder if this distinction is useful though.

Given your proposed change below, the redundancy between BB_SLICE_BUFFER and COMPRESSED_NONE is removed anyway.